05-recursive-resources/proposal.md

```diff
@@ -0,0 +1,21 @@
+# Summary
+
+Merging resource types into resources:
+
+* Easily introduces visibility into resource types (storing history of versions, pinning a version, etc).
+
+* Removes the need for duplicate code (example, supporting fly check-resource-type and fly check-resource).
+
+* Allows for easier process for creating a new resource type (can create a pipeline that runs tests on new versions of a resource type)
+
+
+
+# Proposal
+
+Currently, Concourse treats resources and resource types quite similarly. This proposal is to discuss the possibility of merging them together into just resources. This will end up with resource_types being defined as a resource in the pipeline and removing the resource_types section entirely. A resource type, in result, will just become a resource that will be used by another resource as its type (which will introduce recursive resources).
+
```

Comment:
I suggest to add a new section here that makes an example: an excerpt of the current configuration and how it would become with this proposal.